build.sh

```diff
@@ -21,28 +21,28 @@ fi
 # Install the Dashboard dependencies, including the linter's dependencies and
 # the unit test dependencies.  First, remove any existing Node modules and
 # package-lock.json to ensure that we install the latest.
```

Comment:
I guess we can remove this comment from here and move it to the makefile where we are using package-lock.json, right?

Reply:
I sympathize with what you are trying to say, but I don't think it works.  Inside the context of the Makefile, the targets `clean` and `node_modules` do specific things which, in that context, don't have so much to do with "installing dependencies" and determining "the latest":  there they are simply removing certain artifacts and recreating them.  From the perspective _outside_ the Makefile, that (re)creation results in installing the dependencies and doing so using the latest versions.

I could add a comment at the Makefile target which invokes `npm install` that tries to explain that it will install the stuff requested by `package.json` (i.e., our dependencies) and that it will follow the constraints in the `package-lock.json` file if it exists and that it will create the file if does not, but that's just a description of what `npm install` does (and it's mostly evident from the construction of the rule), so I don't think it would really add anything there.  (Likewise, I could add a comment to the `clean` target which points out that removing the `package-lock.json` file will result in an install of the latest versions on the _next_ invocation of the `node_modules` target, but that's not quite the right spot for a comment like that, not to mention that it  should be sort of obvious.)  So, I don't think there's much benefit to trying to add comments to the Makefile.

As for `build.sh`, the comment above describes the impact which invoking the `clean` target and the `node_modules` targets has in _this_ context, which isn't really obvious from the command invocation.  The token "node_modules" does not say (to me) much about "dependencies" and it says nothing about "latest".  So, I think having this comment here is worthwhile.